bee-api/bee-rest-api/src/types/dtos.rs

```diff
@@ -536,6 +675,497 @@ pub struct ReferenceUnlockBlockDto {
     pub index: u16,
 }
 
+/// Points to the unlock block of a consumed alias output.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct AliasUnlockBlockDto {
+    #[serde(rename = "type")]
+    pub kind: u8,
+    #[serde(rename = "reference")]
+    pub index: u16,
+}
+
+/// Points to the unlock block of a consumed NFT output.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct NftUnlockBlockDto {
+    #[serde(rename = "type")]
+    pub kind: u8,
+    #[serde(rename = "reference")]
+    pub index: u16,
+}
+
+/// Describes an extended output.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ExtendedOutputDto {
+    #[serde(rename = "type")]
+    pub kind: u8,
+    // Deposit address of the output.
+    pub address: AddressDto,
+    // Amount of IOTA tokens held by the output.
+    pub amount: u64,
+    // Native tokens held by the output.
+    #[serde(rename = "nativeTokens")]
+    pub native_tokens: Vec<NativeTokenDto>,
+    #[serde(rename = "blocks")]
+    pub feature_blocks: Vec<FeatureBlockDto>,
+}
+
+impl From<&ExtendedOutput> for ExtendedOutputDto {
+    fn from(value: &ExtendedOutput) -> Self {
+        Self {
+            kind: ExtendedOutput::KIND,
+            address: value.address().into(),
+            amount: value.amount(),
+            native_tokens: value.native_tokens().iter().map(Into::into).collect::<_>(),
+            feature_blocks: value.feature_blocks().iter().map(Into::into).collect::<_>(),
+        }
+    }
+}
+
+impl TryFrom<&ExtendedOutputDto> for ExtendedOutput {
+    type Error = Error;
+
+    fn try_from(value: &ExtendedOutputDto) -> Result<Self, Self::Error> {
+        let mut builder = ExtendedOutputBuilder::new((&value.address).try_into()?, value.amount);
+        for t in &value.native_tokens {
+            builder = builder.add_native_token(t.try_into()?);
+        }
+        for b in &value.feature_blocks {
+            builder = builder.add_feature_block(b.try_into()?);
+        }
+        Ok(builder.finish()?)
+    }
+}
+
+/// Describes a native token.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct NativeTokenDto {
+    // Identifier of the native token.
+    pub token_id: TokenIdDto,
+    // Amount of native tokens.
+    pub amount: U256Dto,
```

Comment:
Some `Dto`'s have comments and some don't. Maybe we should unify? If we keep them, these should probably be rustdoc comments.